stock_location_tank/models/stock_location.py

```diff
@@ -0,0 +1,169 @@
+# Copyright NuoBiT Solutions - Frank Cespedes <fcespedes@nuobit.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+from odoo.tools.safe_eval import json
+
+
+class Location(models.Model):
+    _inherit = "stock.location"
+
+    is_tank = fields.Boolean()
+    is_blocked = fields.Boolean(compute="_compute_is_blocked")
+    capacity = fields.Float()
+    allowed_uom_id = fields.Many2one(comodel_name="uom.uom", string="Unit of Measure")
+    # S'hauria d'amagar el sequence_id si el use_provider_lot_number es True
+    sequence_id = fields.Many2one(comodel_name="ir.sequence")
+    allowed_product_ids = fields.Many2many(
+        comodel_name="product.product", string="Allowed Products"
+    )
+    production_id = fields.Many2one(
+        comodel_name="mrp.production", compute="_compute_production_id"
+    )
+    capacity_occupied = fields.Float(compute="_compute_capacity_occupied")
+    # Mirar el tema de perque la string esta " " mira el percentpie
+    percentage_occupied = fields.Float(compute="_compute_percentage_occupied")
+    use_provider_lot_number = fields.Boolean()
```

Comment:
Negate the logic in order to avoid using "provider" and making it more generic. Something like `create_new_lot`